File: valid-palindrome/valid-palindrome.py
def is_palindrome(s):
    text = [c.lower() for c in s if c.isalpha()]
    start = 0
    end = len(text) - 1
    while start < end:
        if text[start] != text[end]:
            return False
        start += 1
        end -= 1
    return True

File: valid-palindrome/valid-palindrome.js
function isPalindrome(s) {
  const text = [];
  for (let i = 0; i < s.length; i++) {
    if (s[i].match(/^[a-z0-9]+$/i)) {
      text.push(s[i].toLowerCase());
    }
  }
  let left = 0;
  let right = text.length - 1;
  while (left < text.length) {
    if (text[left] !== text[right]) {
      return false;
    }
    left++;
    right--;
  }
  return true;
}
